great_expectations/util.py

```diff
@@ -1709,14 +1711,12 @@ def convert_ndarray_float_to_datetime_tuple(
     return tuple(convert_ndarray_float_to_datetime_dtype(data=data).tolist())
 
 
-def is_ndarray_decimal_dtype(
-    data: "npt.NDArray",
-) -> TypeGuard["npt.NDArray"]:
+def is_ndarray_decimal_dtype(data: np.ndarray) -> bool:
```

Comment:
The use of TypeGuard was correct. Can you please explain the issue you are having with it?

Reply:
@Kilo59 Is it still correct/appropriate/useful, given that I had to change the logic from `all()` to `any()`?  Thanks.

Reply:
Np Arrays are all of the same type, so if any datatype is a decimal they all are correct?

Reply:
@Kilo59 There are exceptions, in which case, the type is `dtype.object` -- which can happen.  This is exactly what happens in our case:
```
[Decimal('0.0') 2.5] ; TYPE: <class 'numpy.dtype[object_]'>
```
because we have in the database, the occurrence of `Decimal` and `Float` in the same column.  Hence, this subtlety had to be handled.  (Fortunately, both are `SupportsFloat` type).  Thanks.

Reply:
This answers my question below. Instead of testing for `any` would it be better to test that they are all one of these types?

Reply:
That is you can use `all` and pass a list of types to the `isinstance` check?

Reply:
Were you getting any kind of typing errors before you changed this?

`TypeGuard` doesn't affect runtime behavior at all, but it's possible it was making the typing system think that if this returned false that the type wasn't an `NDArray` at all 🤔 

https://mypy.readthedocs.io/en/stable/type_narrowing.html#user-defined-type-guards

Reply:
@Kilo59 There were no errors -- I just saw the case of an increased complexity without some kind of an example (or a test) to justify it, so the benefit was unclear to me.  But after our discussion, it seems that the real benefit will come in the future, so I brought it back. :-)